gun/phttp/base.go

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) 2017 Yandex LLC. All rights reserved.
+// Use of this source code is governed by a MPL 2.0
+// license that can be found in the LICENSE file.
+// Author: Vladimir Skipor <skipor@yandex-team.ru>
+
+package phttp
+
+import (
+	"context"
+	"io"
+	"io/ioutil"
+	"log"
+	"net/http"
+
+	"github.com/facebookgo/stackerr"
+
+	"github.com/yandex/pandora/aggregate"
+	"github.com/yandex/pandora/ammo"
+	"github.com/yandex/pandora/gun"
+)
+
+// TODO: inject logger
+type Base struct {
+	Do      func(r *http.Request) (*http.Response, error) // Required.
+	Connect func(ctx context.Context) error               // Optional hook.
+	Results chan<- *aggregate.Sample                      // Lazy set via BindResultTo.
+}
+
+var _ gun.Gun = (*Base)(nil)
+
+func (b *Base) BindResultsTo(results chan<- *aggregate.Sample) {
+	if b.Results != nil {
+		log.Panic("already binded")
+	}
+	if results == nil {
+		log.Panic("nil results")
+	}
+	b.Results = results
+}
+
+// Shoot to target, this method is not thread safe
```

Comment:
It will iff Do and Connect are thread safe. I will update doc in next commit.
But anyway thread safe is not required in current architecture: User make requests consistently and synchronous.